### PR TITLE
Show accessible tables only in New Query view for PostgreSQL

### PR DESCRIPTION
**Pull request by shinsuke-nara**
## What type of PR is this? (check all applicable)

- [x] Bug Fix

#### Background

Left pane of `New Query` view shows table list. The list includes tables which user can not access.
Tables which user can not access should be hidden.

#### The bug

Function `redash.query_runner.pg.PostgreSQL._get_tables` does not consider privileges of a user for tables.

#### The Fix

Get tables for which `current_user` of PostgreSQL have privileges. 

## Related Tickets & Documents

--

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

--

### CODE REVIEW

**Comment by ranbena** [comment]
@arikfr is omitting inaccessible tables the desired user experience?
An alternative would be to show these in disabled mode with a tooltip explanation, but I'm not knowledgable enough to know if this has any value.

Also, should this type of filtering be replicated across all query runners?

**Comment by arikfr** [comment]
@ranbena 

> is omitting inaccessible tables the desired user experience?

Yes.

> Also, should this type of filtering be replicated across all query runners?

In theory query runners should be implemented in this way already. But there is no single recipe that will work for all of them anyway.

**Comment by shinsuke-nara** [comment]
@arikfr 

Thank you for checking my PR.
I agree with you. I will fix this PR with option 3.

**Comment by shinsuke-nara** [comment]
@arikfr 

I fixed this PR according to your advice.
Could you check it if you have time?

File: redash/query_runner/pg.py [hunk 1]
```diff
@@ -126,33 +126,12 @@ def _get_definitions(self, schema, query):
...
-        JOIN pg_attribute a
-        ON a.attrelid = c.oid
-        AND a.attnum > 0
-        AND NOT a.attisdropped
-        WHERE c.relkind IN ('r', 'v', 'm', 'f', 'p')
+        SELECT table_schema,
+               table_name,
+               column_name
+        FROM information_schema.columns
+        WHERE table_schema NOT IN ('pg_catalog', 'information_schema')
```
Comment (arikfr) [inline redash/query_runner/pg.py]: Thank you for the update, but we still need the old query for the "special types", so it should be something like:

```python
query = """
        SELECT s.nspname as table_schema,
               c.relname as table_name,
               a.attname as column_name
        FROM pg_class c
        JOIN pg_namespace s
        ON c.relnamespace = s.oid
        AND s.nspname NOT IN ('pg_catalog', 'information_schema')
        JOIN pg_attribute a
        ON a.attrelid = c.oid
        AND a.attnum > 0
        AND NOT a.attisdropped
        WHERE c.relkind IN ('m', 'f', 'p')

        UNION 

        SELECT table_schema,
               table_name,
               column_name
        FROM information_schema.columns
        WHERE table_schema NOT IN ('pg_catalog', 'information_schema')
"""

Thanks :)

Comment (shinsuke-nara) [reply redash/query_runner/pg.py]: @akariv 

Thank you for the advice. I have fixed again.

**Comment by arikfr** [comment]
Thanks!

**Comment by shinsuke-nara** [comment]
@arikfr 

Thank you for accepting my PR :)

**Pull request merged**